sympy/matrices/matrices.py

```diff
@@ -1130,7 +1130,7 @@ def eigenvals(self, error_when_incomplete=True, **flags):
 
         # make sure the algebraic multiplicty sums to the
         # size of the matrix
-        if error_when_incomplete and sum(m for m in eigs.values()) != self.cols:
+        if error_when_incomplete and type(eigs)==dict and sum(m for m in eigs.values()) != self.cols:
```

Comment:
It is probably preferable to write `isinstance(eigs, dict)`. Then `eigs` could also be in a subclass like `defaultdict`.